### PR TITLE
Dk add tests

### PR DESCRIPTION
**Pull request by DavidKnott**
Fix password test failure in the repo's controller
Add has secure password 
Create a test for the download feature
Change home index view from aws url to repo name

### CODE REVIEW

**Review by ethanbennett — APPROVED** [review]
Nice fixes!

File: spec/requests/downloads_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+require 'rails_helper'
+
+describe "DownloadsController" do
+  it "can download a repo" do
```
Comment (Laszlo-JFLMTCO) [inline spec/requests/downloads_controller_spec.rb:4]: @DavidKnott Great job finding a solution with VCR!!!

File: spec/features/users/user_can_see_nested_content_spec.rb [hunk 2]
```diff
@@ -9,6 +9,11 @@
 
     expect(current_path).to eq home_index_path
     expect(page).to have_content "#{user.first_name} #{user.last_name}"
-    #add tests
+    expect(page).to have_content user.repos.first.name
```
Comment (Laszlo-JFLMTCO) [inline spec/features/users/user_can_see_nested_content_spec.rb:12]: Tests, yay!!!! :-)

File: spec/features/users/user_can_see_root_content_spec.rb [hunk 3]
```diff
@@ -10,6 +10,8 @@
 
     expect(current_path).to eq home_index_path
     expect(page).to have_content "#{user.first_name} #{user.last_name}"
-    #add tests
+    expect(page).to have_content user.repos.first.name
```
Comment (Laszlo-JFLMTCO) [inline spec/features/users/user_can_see_root_content_spec.rb:13]: Tests, yay!!!! :-)

**Comment by Laszlo-JFLMTCO** [comment]
@david Good job! Ready to merge it!

**Pull request merged**